core/src/main/java/org/bitcoinj/core/CheckpointManager.java

```diff
@@ -144,11 +146,11 @@ private Sha256Hash readBinary(InputStream inputStream) throws IOException {
                     throw new IOException("Incomplete read whilst loading checkpoints.");
                 StoredBlock block = StoredBlock.deserializeCompact(params, buffer);
                 ((Buffer) buffer).position(0);
-                checkpoints.put(block.getHeader().getTimeSeconds(), block);
+                checkpoints.put(block.getHeader().getTimeInstant(), block);
             }
             Sha256Hash dataHash = Sha256Hash.wrap(digest.digest());
             log.info("Read {} checkpoints up to time {}, hash is {}", checkpoints.size(),
-                    TimeUtils.dateTimeFormat(checkpoints.lastEntry().getKey() * 1000), dataHash);
+                    TimeUtils.dateTimeFormat(checkpoints.lastEntry().getKey().toEpochMilli()), dataHash);
```

Comment:
We should (separately) add a `TimeUtils.dateTimeFormat(Instant)`

Reply:
Yes, I have that commit on a branch already, but decided to PR it later.